src/Domain/Insights/ForbiddenSecurityIssues.php

```diff
@@ -39,48 +42,75 @@ public function getDetails(): array
         }
 
         try {
-            $issues = json_decode((string) $this->getResult(), true);
+            $this->getResult();
         } catch (InternetConnectionNotFound $exception) {
             self::$details = [
                 Details::make()->setMessage($exception->getMessage()),
             ];
             return self::$details;
         }
 
-        if ($issues === null) {
-            return [];
+        return self::$details;
+    }
+
+    private function getResult(): void
+    {
+        $composerPath = sprintf('%s/composer.lock', $this->collector->getDir());
+
+        if (! file_exists($composerPath)) {
+            throw new ComposerNotFound('composer.lock not found. Try launch composer install');
         }
 
+        $packages = json_decode(file_get_contents($composerPath), true);
+        $packagesToCheck = array_combine(
+            array_map(static function (array $detail): string {
+                return $detail['name'];
+            }, $packages['packages']),
+            array_map(static function (array $detail): string {
+                return $detail['version'];
+            }, $packages['packages'])
+        );
+
+        $advisoryList = $this->retrieveAdvisoriesListForPackages(array_keys($packagesToCheck));
+
         self::$details = [];
-        foreach ($issues as $packageName => $package) {
-            foreach ($package['advisories'] as $advisory) {
-                self::$details[] = Details::make()->setMessage(
-                    "${packageName}@{$package['version']} {$advisory['title']} - {$advisory['link']}"
-                );
-            }
+        $packagesToCheck = array_filter(
+            $packagesToCheck,
+            static function (string $packageName) use ($advisoryList): bool {
+                return \array_key_exists($packageName, $advisoryList['advisories']);
+            },
+            ARRAY_FILTER_USE_KEY
+        );
+
+        foreach ($packagesToCheck as $packageName => $version) {
+            $issues = $advisoryList['advisories'][$packageName];
+
+            $this->addIssuesDetails($packageName, $version, $issues);
         }
+    }
 
-        return self::$details;
+    private function retrieveAdvisoriesListForPackages(array $packagesName): array
+    {
+        $client = HttpClient::createForBaseUri(self::PACKAGIST_ADVISORIES_URL);
+        // TODECIDE: Implement caching on this ?
+        // $client = new CachingHttpClient($client, new Store(sys_get_temp_dir() . DIRECTORY_SEPARATOR . 'phpinsights'));
```

Comment:
I let this commented for now. In 1.14, we don't have yet a cache mechanism, so there is no easy way for user to clear this cache.